setting state.on to the right value after sending state changes to the bridge #38

willie68 opened the pull request:
THis is the simple solution. At the moment the response of the bridge will not be evaluated. (Maybe in the next step)

amimof commented:
@willie68 Thanks for contributing 👏 Please tell me when this is ready for review

willie68 commented:
I think review can happen now. 
I don't know why travis can't build this...

amimof commented:
How about if `Bri` is set to `0`, i guess the state.on would always be true in any case?

amimof commented on group.go:
```diff
@@ -198,6 +203,27 @@ func (g *Group) CtContext(ctx context.Context, new uint16) error {
+	return g.ColContext(context.Background(), new)
+}
+
+// CtContext sets the light color temperature state property
```
Should be `ColContext` instead of `CtContext`

amimof replied:
Needs tests

amimof commented on light.go:
```diff
@@ -246,3 +276,31 @@ func (l *Light) AlertContext(ctx context.Context, new string) error {
 	return nil
 }
+
+func ConvertRGBToXy(newcolor color.Color) ([]float32, uint8) {
```
Needs doc comment. See other public functions for reference.

amimof replied:
Please add a unit test for this function

amimof commented on light.go:
```diff
@@ -190,6 +198,27 @@ func (l *Light) CtContext(ctx context.Context, new uint16) error {
+	return l.ColContext(context.Background(), new)
+}
+
+// CtContext sets the light color temperature state property
```
Should be `ColContext` instead of `CtContext`

amimof replied:
Also needs unit testing

amimof commented on group.go:
```diff
@@ -198,6 +203,27 @@ func (g *Group) CtContext(ctx context.Context, new uint16) error {
+}
+
+// CtContext sets the light color temperature state property
+func (g *Group) ColContext(ctx context.Context, new color.Color) error {
```
Needs tests

amimof commented on light.go:
```diff
@@ -190,6 +198,27 @@ func (l *Light) CtContext(ctx context.Context, new uint16) error {
+}
+
+// Ct sets the light color temperature state property
+func (l *Light) Col(new color.Color) error {
```
Needs tests

amimof commented on light.go:
```diff
@@ -190,6 +198,27 @@ func (l *Light) CtContext(ctx context.Context, new uint16) error {
+	return nil
+}
+
+// Ct sets the light color temperature state property
```
Fix this comment

amimof commented on group.go:
```diff
@@ -198,6 +203,27 @@ func (g *Group) CtContext(ctx context.Context, new uint16) error {
+	return nil
+}
+
+// Ct sets the light color temperature state property
```
Update comment

amimof commented on group.go:
```diff
@@ -198,6 +203,27 @@ func (g *Group) CtContext(ctx context.Context, new uint16) error {
+}
+
+// Ct sets the light color temperature state property
+func (g *Group) Col(new color.Color) error {
```
Needs tests

willie68 commented:
Changed the comments, added comment on ConvertRGB, adding tests.
Beside the ConvertRGBToXy Test, all tests are failing on my side. 
(Maybe because the Hue is not responding, because it's off. :-) )

amimof commented:
@willie68 Did you read my latest comment about `Bri` being `0` and what the expected behavior would be?

willie68 commented:
I just tested with bri = 0 and the light are not off, it's the lowest possible brightness but it's not off. So setting lights to bright 0 will not shut them down. That means state.on = true even if the bri is 0.

amimof commented on light_test.go:
```diff
@@ -341,6 +371,7 @@ func TestSetLightAlert(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
+	assert.True(t, light.IsOn())
```
This line is why tests fails. It doesn't receive expected result. Can you have a look?

willie68 commented:
Just missed a single light.State.On = true
